base/special/gamma.jl

```diff
@@ -482,3 +482,7 @@ eta(x::Integer) = eta(Float64(x))
 eta(x::Real)    = oftype(float(x),eta(Float64(x)))
 eta(z::Complex) = oftype(float(z),eta(Complex128(z)))
 @vectorize_1arg Number eta
+
+Base.binomial{S<:Number,T<:Number}(n::S, k::T) = ( (isinteger(n) && isinteger(k))
```

Comment:
here with `isinteger` I don't check on type but only if I can cast to some integer type without loss. This is to be able to compute `binomial(2.0, 1)`